api/src/main/java/org/openmrs/api/ProgramWorkflowService.java

```diff
@@ -530,4 +533,60 @@ public void purgeConceptStateConversion(ConceptStateConversion conceptStateConve
 	 * @should throw an error when multiple program states with same uuid are found
 	 */
 	public ConceptStateConversion getConceptStateConversionByUuid(String uuid);
+
+	/**
+	 * Fetches a list of all ProgramAttributeTypes
+	 *
+	 * @return list of ProgramAttributeType objects
+	 */
+	@Transactional(readOnly = true)
+    @Authorized({"View PatientProgram Attribute Types"})
+    List<ProgramAttributeType> getAllProgramAttributeTypes();
```

Comment:
I know they forgot this in Bahmni, but you need to add a `getAllProgramAttributeTypes()` method that takes in boolean argument that specifies whether retired ones should be included or excluded. In theory, this method that has no arguments should delegate to the other with a value of false. I personally prefer naming these methods `getProgramAttributeTypes` rather than `getAllProgramAttributeTypes` but I guess most existing methods are named that way so I guess it's fine.